### ISSUE TITLE
v3: `helm template` is missing the flag to render only selected templates

### ISSUE THREAD

**Issue by mumoshu** [helm/helm#6633]
In helm v2 we had `helm template mychart-1.2.3.tgz -x templates/your.yaml` that can be used to render only selected template files in the chart.

If you run `helm template --help`, this is featured in the beginning of the output like:

```
To render just one template in a chart, use '-x':

	$ helm template mychart -x templates/deployment.yaml

Usage:
  helm template [flags] CHART

Flags:
  -x, --execute stringArray      only execute the given templates
```

This feature isn't announced for removal so I'm assuming this is a regression in v3. Can we just reimplement it for v3?

If we are to reimplement this for helm v3, a concern is that `helm template` now supports templating the remote chart directly, without firstly `helm fetch`ing the chart. The v3 version of the feature should just work regardless of the chart is local or remote.

---

Output of `helm version`: helm-3.0.0-beta.1~4

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Any

**Comment by rimusz** [comment]
yes, it is the regression, so it can be reimplement in v3

**Comment by bacongobbler** [comment]
It is NOT a regression in the sense that it was forgotten.

The -x flag was intentionally removed from Helm 3 as it would introduce a problem with the Lua hook system or library charts: if a template relied on a Lua hook being executed, the -x flag would prevent that hook from running first. That is not a good design.

What we wanted to do post-3.0 is to re-implement -x as a selective post-render step, where it would select certain templates based on a label selector. That way, the chart goes through its full execution and you still receive a single template post-render.

Does that make sense?

**Comment by bacongobbler** [comment]
closed via #6634

**Comment by fjolublar** [comment]
Should the helm2-to-3 migration document be adapted to be more true to the change:

https://helm.sh/docs/topics/v2_v3_migration/

```
template: -x/--execute argument renamed to -s/--show-only
```

This reads like the logic for both the flags is the same, but it's not.

`-x` in helm2 was doing single-file template, but `-s` is doing template on all chart and only **showing** the file given.
